### ISSUE TITLE
[BUG] Gradient accumulation logic in train_examples/ might be incorrect?

### ISSUE THREAD

**Issue by hucorz** [meituan-longcat/LongCat-Image#46]
Hi, thanks for the great work!
While reviewing the training scripts (specifically in train_examples/), I felt the gradient accumulation logic seemed a bit unusual compared to standard implementations. Of course, this is fine when default gradient_accumulation_steps=1, but it might be problematic when accumulation is enabled.
I asked gemini to analyze the code, and here is the output:

Gemini 3 Pro Analysis:
There are likely two issues with the current implementation when `gradient_accumulation_steps > 1`:
1. `optimizer.zero_grad()` clears accumulated gradients prematurely
In the current code:
```python
with accelerator.accumulate(model):
    optimizer.zero_grad()  # <--- Clears gradients at the start of the step
    # ... forward & backward ...
    optimizer.step()
```
When accelerator syncs gradients (on the final accumulation step), calling zero_grad() at the beginning of the block will wipe out all the gradients accumulated from previous micro-batches. This effectively disables gradient accumulation, as the optimizer step will only see the gradients from the very last micro-batch.
Recommendation: Move `optimizer.zero_grad()` to after `optimizer.step()`.
2. Loss is not scaled manually
Accelerate handles the gradient synchronization, but it does not automatically scale the loss during the backward pass.
Recommendation: You should divide the loss by the accumulation steps:
```python
accelerator.backward(loss / args.gradient_accumulation_steps)
```
Without this, the effective gradients (summed over steps) will be much larger than expected, effectively increasing the learning rate by a factor of `N`.
Could you please confirm if my understanding is correct? Thanks

**Comment by hucorz** [comment]
Additional finding regarding the LR Scheduler:

I also noticed that the scheduler initialization in train_edit_dpo.py seems incorrect for multi-GPU training:

```python
num_warmup_steps=args.lr_warmup_steps * accelerator.num_processes,
num_training_steps=args.max_train_steps * accelerator.num_processes,
```

Multiplying by num_processes here causes the scheduler to expect a much longer training duration than reality (since the training loop breaks at args.max_train_steps regardless of the number of GPUs). This effectively prevents the learning rate from decaying correctly by the end of training. It should likely be just args.max_train_steps.
